gradle-diff-task-dependencies/sample/build.gradle

```diff
@@ -0,0 +1,27 @@
+import com.novoda.gradle.diffdependencytask.DiffDependencyTaskPlugin
+
+apply plugin: 'java'
+apply plugin: DiffDependencyTaskPlugin
+
+diffDependency {
+    // custom changed files provider available to test in non-CI environments
+    changedFilesProvider = {
+        return ['apple/fastlane', 'android/build.gradle']
+    }
+}
+
+task('testAndroid') << {
+    println 'This is an android test task.'
+}
+
+task('testApple') << {
+    println 'This is an apple test task.'
+}
+
+task('selectiveTest') << {
+    println 'This will run after 0 or more between "testAndroid" and "testApple".'
+}
+
+tasks.getByName('selectiveTest')
+        .ifDiffMatches(~'android/.*').dependsOn('testAndroid')
```

Comment:
I'd rather have a well defined API over:
* named containers on `ext` (so that the dependency declaration is similar to the regular one)
* `dependsOn` with patterns and task names (as the original `dependsOn` already uses varargs of `Object`, discriminating between the two could be messy if possible at all)

These were really the only two reasons that pushed me to go for the "code over configuration" approach.